design/k8s-job-hook.md

```diff
@@ -0,0 +1,176 @@
+# Kubernetes Job Hook Proposal
+
+## Goals
+
+- Enable hooks that trigger execution of some behavior directly by the lakeFS server using Kubernetes
+- Enable non-blocking execution for post-merge and post-merge
+- Support successful job completion as condition for the commit and merge operations
+
+
+## Non Goals
+
+- GitHub Actions - fully maintained, virtualized environment with pre-build images to execute pre/post hooks
+
+
+## Proposition
+
+### Overview
+
+New lakeFS hook type, triggered by pre/post commit/merge events to execute jobs on k8s cluster.
+
+The user will provide a hook definition that will include the name of the image (docker image) and arguments that will be used for the hook.
+While commit/merge occurs, lakeFS will request to execute the job definition on the cluster and wait (optional) for success in order to complete the operation.
+
+### New hook definition
+
+Based on the current actions mechanism, the user will need to write a yaml file and upload it into the repository `_lakefs_actions` folder.
+
+Example of an action using the new hook definition:
+
+```yaml
+name: Branch version tagger
+description: set version tag on each merge to main
+on:
+  post-merge:
+    branches:
+      - main
+hooks:
+  - id: update_tag
+    type: k8s-job
+    description: Create a tag based on last version
+    properties:
+      image: "registry/lakefs-hooks:4"
+      command: ["python"]
+      args: ["bump-version.py"]
+      env:
+      - name: REPOSITORY
+        value: customers
+      - name: PROJECT
+        value: alpha
+```
+
+In this example we specified a post merge hook to execute a job. The job will use the image `next/lakefs-hooks:4` with the command python with the argument `bump-version.py`.
```

Comment:
It should be super clear that:
1. `registry/lakefs-hooks:4` is defined and managed by the user. The name is misleading and may suggest we maintain it.
2. Same goes to `bump-version.py`.

Reply:
1. Update to 'myregistry/myhook:4' and removed any NEXT from the paragraph.
2. Added that his is a user-supplied image.
3. The bump version is related to next - but it is a valid use-case we can describe.